pkg/controller/configmap/config.go

```diff
@@ -733,3 +812,86 @@ func IsMTUChanged(currConfigMap *corev1.ConfigMap, prevConfigMap *corev1.ConfigM
 		return true
 	}
 }
+
+func FillNsxAuthCfg(configmap *corev1.ConfigMap, nsxSecret *corev1.Secret) error {
+	errs := []error{}
+	data := &configmap.Data
+	cfg, err := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+	if err != nil {
+		log.Error(err, "failed to load ConfigMap")
+		return err
+	}
+
+	if nsx_cert, found := nsxSecret.Data["tls.crt"]; found {
```

Comment:
So if the "tls.crt" was not found in nsxSecret, i.e. found is false, we don't need to do anything, right?

Reply:
yes, I think so. In this case, user should use adim/password for auth, of course, these should not in production env, might be in test env.

Reply:
Then can we simplify L793-L800 to:
```
	if nsx_cert, found := nsxSecret.Data["tls.crt"]; found && len(nsx_cernt) > 0 {
		appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "nsx_api_cert_file", "/etc/nsx-ujo/nsx-cert/tls.crt", true))
	}
```
?
Because I think `None` is the default value so that we don't need to set it explicitly?

Reply:
I tested again. I agreed with you advice.  we don't need to set it explicitly when secret field is empty.  if secret is empty, this option should not be set by user as well. otherwise, there will be some issue. So, we just need to check the case that secret is created, and we set it for users. This will be safe for both fresh installation and upgrade